client/pages/all-records.jsx

```diff
@@ -28,62 +32,139 @@ class AllRecords extends React.Component {
       .catch(err => console.error(err));
   }
 
+  editRecord(record, recordIndex) {
+    this.setState({
+      recordToEdit: `${record.datePerformed} ${recordIndex}`,
+      editRecordName: record.names[recordIndex],
+      editRecordCost: record.cost[recordIndex]
+    });
+  }
+
+  handleChange(event) {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  handleSubmit(event, record, recordIndex, accIndex) {
+    event.preventDefault();
+    const { editRecordName, editRecordCost, records } = this.state;
+    const updatedRecords = {
+      date: record.datePerformed,
+      oldName: record.names[recordIndex],
+      oldCost: record.cost[recordIndex],
+      newName: editRecordName,
+      newCost: editRecordCost
+    };
+    const newRecords = [...records];
+    const { names, cost } = newRecords[accIndex];
+    newRecords[accIndex].total = (parseInt(records[accIndex].total) - parseInt(record.cost[recordIndex])) + parseInt(editRecordCost);
+    names[recordIndex] = editRecordName;
+    cost[recordIndex] = editRecordCost;
+    this.setState({
+      records: newRecords,
+      recordToEdit: null
+    });
```

Comment:
I recommend doing this state update when your fetch() actually succeeds, _i.e._ in the second .then() callback